Android/app/src/main/java/app/intra/sys/firebase/AnalyticsWrapper.java

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright 2019 Jigsaw Operations LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package app.intra.sys.firebase;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import app.intra.sys.firebase.BundleBuilder.Params;
+import com.google.firebase.analytics.FirebaseAnalytics;
+
+/**
+ * Library of Analytics events reported by Intra.  In addition to improving the reporting API,
+ * this class also attaches the Mobile Country Code (MCC) to each event, which indicates the client
+ * country, to compensate for country classification errors in Firebase.
+ */
+public class AnalyticsWrapper {
+
+  // Analytics events
+  private enum Events {
+    BOOTSTRAP,
+    BOOTSTRAP_FAILED,
+    BYTES,
+    EARLY_RESET,
+    STARTVPN,
+    TLS_PROBE,
+    TRY_ALL_ACCEPTED,
+    TRY_ALL_CANCELLED,
+    TRY_ALL_DIALOG,
+    TRY_ALL_FAILED,
+    TRY_ALL_REQUESTED,
+    UDP,
+  }
+
+  // Mobile Country Code, or 0 if it is unknown.
+  private final int mcc;
+  private final FirebaseAnalytics analytics;
+  private AnalyticsWrapper(int mcc, FirebaseAnalytics analytics) {
```

Comment:
I think we should move to ISO country codes instead, since MCC is a pain to convert.

Reply:
Done